controllers/memberoperatorconfig/configuration.go

```diff
@@ -17,67 +20,111 @@ const (
 var logger = logf.Log.WithName("configuration")
 
 type Configuration struct {
-	m       *toolchainv1alpha1.MemberOperatorConfigSpec
+	cfg     *toolchainv1alpha1.MemberOperatorConfigSpec
 	secrets map[string]map[string]string
 }
 
+// GetConfiguration returns a Configuration using the cache, or if the cache was not initialized
+// then retrieves the latest config using the provided client and updates the cache
+func GetConfiguration(cl client.Client) (Configuration, error) {
+	config, secrets, err := commonconfig.GetConfig(cl, &toolchainv1alpha1.MemberOperatorConfig{})
+	if err != nil {
+		// return default config
+		logger.Error(err, "failed to retrieve Configuration")
+		return Configuration{cfg: &toolchainv1alpha1.MemberOperatorConfigSpec{}}, err
+	}
+	return newConfiguration(config, secrets), nil
+}
+
+// GetCachedConfiguration returns a Configuration directly from the cache
+func GetCachedConfiguration() Configuration {
+	config, secrets := commonconfig.GetCachedConfig()
+	return newConfiguration(config, secrets)
+}
```

Comment:
As in the other repos, I would really prefer having and using only one function.

Reply:
Agreed, I did notice that and planned to change it but I was planning to do it in a separate PR if that's alright with you.

Reply:
ok, I'm fine with it :+1: